### PR TITLE
Model loading from Stream

### PR DESCRIPTION
**Pull request by amironoff**
## Summary

It's not always convenient to store models in the file system. Sometimes, it makes more sense to have them stored in the database, embedded resources or elsewhere. This PR adds support for loading models from Stream to support these use cases.

## Main changes

- ModelReader modified to accept a model-resolving Func<string, Stream> delegate
- Basic unit tests that check this model loading approach works.

### CODE REVIEW

File: Core/CRFSharp/encoder/ModelWriter.cs [hunk 1]
```diff
@@ -5,24 +5,32 @@
...
 {
-    public class ModelWritter : BaseModel
+    public class ModelWriter : BaseModel
     {
+        private readonly string modelFileName;
+
         private readonly Pool<StringBuilder> _buildersPool =
             new Pool<StringBuilder>(p => new StringBuilder(100), b => b.Clear());
 
+        private readonly ModelReader modelReader;
```
Comment (zhongkaifu) [inline Core/CRFSharp/encoder/ModelWriter.cs]: Thanks for your updating. Why ModelWriter class keeps a modelReader as field ?

Comment (zhongkaifu) [reply Core/CRFSharp/encoder/ModelWriter.cs]: If modelReader is used for incremental training, A new class named ModelReaderWriter or ModelStream would be a better name.

**Comment by zhongkaifu** [comment]
Hi @amironoff ,

Thanks for your contribution. It's great updating for CRFSharp. I have left some comments in your code change, could you please take a look ? In addition, I found build is broken in your change list, could you please take a look as well ?

**Comment by amironoff** [comment]
Hi @zhongkaifu! Oops, looks like some case of "Works on my machine" :) I'll go over the issues and will get back to you later.

File: Core/CRFSharp/encoder/ModelWriter.cs (hunk 1 above)
Comment (amironoff) [reply Core/CRFSharp/encoder/ModelWriter.cs]: ModelReader was historically used in ModelWriter class to read the model for retraining. I just moved it into a class field. Here it is in the original code: https://gist.github.com/amironoff/82ff62db7c9fb112979ff442b4112d1d

On the other hand, you're right, ModelReader isn't always needed here, just when retraining. Therefore, I moved it back into the training method.

**Comment by amironoff** [comment]
@zhongkaifu I updated the pull request and replied to your remark. Build's green. Looks good?

**Comment by zhongkaifu** [comment]
Thanks @amironoff for your contribution.
I have merged it. :)

**Pull request merged**